components/script/script_thread.rs

```diff
@@ -1217,7 +1217,8 @@ impl ScriptThread {
 
         if let Some(root_context) = self.browsing_context.get() {
             for it_context in root_context.iter() {
-                let current_url = it_context.active_document().url().to_string();
+                let doc = it_context.active_document();
+                let current_url = doc.url().to_string();
```

Comment:
This is also for appease the borrow checker, we hold the `document` in temp variable so the Ref<Url> from `.url()` will get out of scope before it. And you just can not place the `Ref<T>` in the return value of the function. See https://github.com/rust-lang/rust/issues/23338